internal/fpath/path.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package fpath
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// FPath creates an OS independently Path Handling Structure
+type FPath struct {
+	local  bool   //set if file is local
+	scheme string //url scheme
+	bucket string //set, when remote scheme
+	path   string //filepath OR remote path (without bucket)
+}
+
+// New creates Struct from handed over URL
+func New(url string) (p FPath, err error) {
+
+	// Check for Schema
+	sepstring := strings.SplitN(url, "://", 2)
+
+	switch len(sepstring) {
+	case 2: // Has Schema
+		p.scheme = sepstring[0]
+		if p.scheme != "sj" {
+			return FPath{}, errors.New("unsupported URL scheme")
+		}
+		//Trim initial / of the path and clean it, afterwards split on first /
+		split := strings.SplitN(path.Clean(strings.TrimLeft(sepstring[1], "/")), "/", 2)
+		p.bucket = split[0]
+		if len(split) == 2 {
+			p.path = filepath.ToSlash(split[1])
+		}
+	case 1: // No Scheme
+		p.local = true
+		p.path = sepstring[0]
+	default: // Everything else is malformed
+		return FPath{}, fmt.Errorf("malformed URL: %s", url)
```

Comment:
Yes, the default case can be removed.

Reply:
It's true, but I would prefer to keep it like this. Otherwise, we will need to satisfy the compiler by adding an artificial return statement in the end of the constructor, which is also confusing.